samples/snippets/csharp/how-to/strings/ModifyStrings.cs

```diff
@@ -0,0 +1,155 @@
+            ReplaceCharArray();
+
+            UnsafeSample();
+        }
+
+        private static void ReplaceCreatesNewString()
+        {
+            // <Snippet1>
+            string source = "The mountains are behind the clouds today.";
+
+            // Replace one substring with another with String.Replace.
+            // Only exact matches are supported.
+            var replacement = source.Replace("mountains", "peaks");
+            Console.WriteLine($"The source string is <{source}>");
+            Console.WriteLine($"The updated string is <{replacement}>");
+            // </Snippet1>
+
+        }
+
+        private static void ReplaceChars()
+        {
+            // <Snippet2>
+            string source = "The mountains are behind the clouds today.";
+
+            // Replace all occurrences of one char with another.
+            var replacement = source.Replace(' ', '_');
+            Console.WriteLine(source);
+            Console.WriteLine(replacement);
+            // </Snippet2>
+        }
+
+        private static void TrimWhitespace()
+        {
+            // <Snippet3>
+            // Remove trailing and leading whitespace.
+            string source = "    I'm wider than I need to be.      ";
+            // Store the results in a new string variable.
+            var trimmedResult = source.Trim();
+            var trimLeading = source.TrimStart();
+            var trimTrailing = source.TrimEnd();
+            Console.WriteLine($"<{source}>");
+            Console.WriteLine($"<{trimmedResult}>");
+            Console.WriteLine($"<{trimLeading}>");
+            Console.WriteLine($"<{trimTrailing}>");
+            // </Snippet3>
+        }
+
+        private static void RemoveText()
+        {
+            // <Snippet4>
+            string source = "Many mountains are behind many clouds today.";
+            // Remove a substring from the middle of the string.
+            string toRemove = "many ";
+            string result = string.Empty;
+            int i = source.IndexOf(toRemove);
+            if (i >= 0)
+            {
+                result= source.Remove(i, toRemove.Length);
+            }
+            Console.WriteLine(source);
+            Console.WriteLine(result);
+            // </Snippet4>
+        }
+
+        private static void ReplaceWithRegEx()
+        {
+            // <Snippet5>
+            string source = "The mountains are still there behind the clouds today.";
+
+            // Use Regex.Replace for more flexibility. 
+            // Replace "the" or "The" with "many" or "Many".
+            // using System.Text.RegularExpressions
+            string replaceWith = "many ";
+            source = System.Text.RegularExpressions.Regex.Replace(source, "the\\s", localReplaceMatchCase, 
+                System.Text.RegularExpressions.RegexOptions.IgnoreCase);
+            Console.WriteLine(source);
+
+            string localReplaceMatchCase(System.Text.RegularExpressions.Match matchExpression)
```

Comment:
Shouldn't local methods start with capital letter as well?
Anyway, a good example of local method usage.